### ISSUE TITLE
Grave not dispawning after claiming items

### ISSUE THREAD

**Issue by Rol4nd58220** [B1n-ry/Youre-in-grave-danger#197]
It happen when the server suddenly close while upon death

**Comment by Rol4nd58220** [comment]
It happen when the server suddenly close while upon death

**Comment by B1n-ry** [comment]
Do you have the server crash log?

**Comment by Rol4nd58220** [comment]
[2024-12-16-1.log](https://github.com/user-attachments/files/18162721/2024-12-16-1.log)

**Comment by Rol4nd58220** [comment]
My bad the server didn't crash but I was timed out from the server but before that I was blown up by a creeper

**Comment by B1n-ry** [comment]
The logs are being spammed from `Netty Epoll Server IO` whatever that is, but other than that there doesn't seem to be anything mentioning that you tried to reclaim the grave. 
Were you able to do so, but the block remained, or did you not get any stuff?

**Comment by Rol4nd58220** [comment]
I was able to claim it but the block remain I did get the stuff

**Comment by B1n-ry** [comment]
Hmm that's odd... I'll try to look into it more, but might be a few days. Have quite a few things to do before christmas. Sorry this issue doesn't quite make the same priority list

**Comment by gyerekecskee** [comment]
It happened to me as well

**Comment by Equinoxx4978** [comment]
Issue for me too

**Comment by B1n-ry** [comment]
Is the issue consistent? Does it happen more than once? Also, can you claim the grave again and get more items? I assume the grave is not breakable still after claiming it?

**Comment by Equinoxx4978** [comment]
> Is the issue consistent? Does it happen more than once? Also, can you claim the grave again and get more items? I assume the grave is not breakable still after claiming it?

The issue isnt consistent, seems to be mostly random.
May be something to do with ping/tps but I'm unsure.

Sometimes they will work perfectly, others it will leave the grave there but give the items. Some of my players have also said they've claimed the grave and gotten nothing back at all.

If you have any ideas about what could cause it i would be happy to try to help

**Comment by B1n-ry** [comment]
Maybe I should have asked before, but what mod loader and mod version are you all using?

**Comment by Equinoxx4978** [comment]
> Maybe I should have asked before, but what mod loader and mod version are you all using?

You're in Grave Danger [Fabric] 2.0.10 on Minecraft 1.20.1

**Comment by B1n-ry** [comment]
Hmmm sorry but I'm pretty lost on how this could happen. I don't think it's a ping issue, as everything is handled server-side. If you get your items back, and the grave is not removed, there should either be an error message printed in chat, or the game just failed to replace the block, which I feel like it shouldn't. I've added a new log message in case that would happen now though.
I have a few things to add still to the next release of the mod. Will probably forget about something though. It should hopefully be out in just a few days, and we'll see if that helps or not.
In case there would happen to be found a way to consistently reproduce the issue though, I'd be very interested to know about it

**Comment by viiizee** [comment]
The players on my server have been having a similar issue on the same version listed above. Sometimes Gravestones don't disappear and they can't collect the item from them; however the experience is returned. I even tried using the command to give back the inventory that was in question and it would do the same thing, only give back experience and nothing else even though I could see the items in the GUI that they lost.

Is it possible that server outages/restarts are causing problems? It seems like it could be that the server restarted both times right after the player died, but I can't seem to get a consensus because by the time I get home to solve the issue for them they've "forgotten" what happened.

**Comment by B1n-ry** [comment]
> The players on my server have been having a similar issue on the same version listed above. Sometimes Gravestones don't disappear and they can't collect the item from them; however the experience is returned. I even tried using the command to give back the inventory that was in question and it would do the same thing, only give back experience and nothing else even though I could see the items in the GUI that they lost.
> 
> Is it possible that server outages/restarts are causing problems? It seems like it could be that the server restarted both times right after the player died, but I can't seem to get a consensus because by the time I get home to solve the issue for them they've "forgotten" what happened.

Have you checked the logs for error messages? To me it seems like there was some error somewhere that wasn't quite severe enough to crash the game, but enough to stop the grave claiming process. In that case I feel like there should have been some error message sent

**Comment by viiizee** [comment]
Yeah I'm sorry I can't be of more help :( I wasn't home when it happened and I can't tell which of these logs would be for You're in Grave Danger. If it happens again I will try to get one for you.
